app/data_processing/calculations.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+from math import sqrt, pow
+
+from app.structures.coords import Coords
+from app.structures.customer_profile import Car, Customer
+from app.structures.shop import Shop
+
+
+def calculating_distances(point1: Coords, point2: Coords) -> float:
+    return sqrt(pow((point1.x - point2.x), 2) + pow((point1.y - point2.y), 2))
+
+
+def ride_expenses(customer: Customer, shop: Shop) -> float:
+    return (customer.car.fuel_consumption
+            * calculating_distances(customer.location, shop.location)
+            * Car.fuel_price / 100)
+
+
+def groceries_cost(customer: Customer, shop: Shop) -> float:
+    return sum([shop.products[item] * quantity
+               for item, quantity in customer.product_cart.items()])
+
+
+def shop_offer(customer: Customer, shop: Shop) -> tuple:
+    trip_shop_home = ride_expenses(customer, shop) * 2
+    cost = round((groceries_cost(customer, shop) + trip_shop_home), 2)
+    return shop, cost
+
+
+def attending_shop(customer: Customer, shop: Shop) -> None:
+    date_of_shopping = datetime(2021, 1, 4, 12, 33, 41)

```

Comment:
The date is hardcoded to a specific value. Consider using `datetime.now()` to get the current date and time dynamically.